storm-core/src/jvm/storm/trident/spout/RichSpoutBatchTriggerer.java

```diff
@@ -56,7 +56,7 @@ public RichSpoutBatchTriggerer(IRichSpout delegate, String streamName, String ba
     @Override
     public void open(Map conf, TopologyContext context, SpoutOutputCollector collector) {
         _delegate.open(conf, context, new SpoutOutputCollector(new StreamOverrideCollector(collector)));
-        _outputTasks = new ArrayList<Integer>();
+        _outputTasks = new ArrayList<>();
```

Comment:
On a different PR someone pointed out that this is syntax required Java 1.7  That seems ok to me since the top level pom.xml specifies 1.7 in the compile plugin, but I wanted to just point that fact out.

I support moving to the diamond notation since it makes the code much cleaner, in my opinion.